bindgen/callbacks.rs

```diff
@@ -231,15 +232,30 @@ pub struct DeriveInfo<'a> {
     pub kind: TypeKind,
 }
 
-/// Relevant information about a type to which new attributes will be added using
+/// Relevant information about an item to which new attributes will be added using
 /// [`ParseCallbacks::add_attributes`].
 #[derive(Debug)]
 #[non_exhaustive]
 pub struct AttributeInfo<'a> {
-    /// The name of the type.
+    /// The name of the item.
     pub name: &'a str,
-    /// The kind of the type.
-    pub kind: TypeKind,
+    /// The kind of the item.
+    pub kind: AttributeItemKind,
+    /// The kind of a method item.
+    pub fn_kind: Option<FunctionKind>,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+/// The kind of the current item.
+pub enum AttributeItemKind {
+    /// The item is a Rust `struct`.
+    Struct,
+    /// The item is a Rust `enum`.
+    Enum,
+    /// The item is a Rust `union`.
+    Union,
+    /// The item is a Rust `fn`.
+    Function,
```

Comment:
How about `Function(FunctionKind)`. This would allow removing `fn_kind` from `AttributeInfo`.

Reply:
You're right, this is a lot nicer. I've made the adjustments as requested.